### PR TITLE
CPDRP-999: Get previous induction data from DQT API

### PR DESCRIPTION
**Pull request by ethanmills**
## Ticket and context

Ticket: CPDPR-999

We should take information on which ECTs have had a previous induction from the DQT API. This informs which participants are eligible for funding.

### Is there anything that the code reviewer should know?
- Removed old temporary DQT API
- changed ineligible participant semantics. I will follow up with a migration to simplify the data stored here to a boolean for previous participation once we're happy this works well

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -678,7 +678,7 @@ DEPENDENCIES
   with_model
 
 RUBY VERSION
-   ruby 3.0.2p107
+   ruby 2.7.2p137
```
Comment (ethanmills) [inline Gemfile.lock:681]: Looks like this got missed before - my mistake

File: app/services/participant_validation_service.rb [hunk 2]
```diff
@@ -19,75 +19,60 @@ def validate
...
   end
 
   def check_first_name_only?
     config[:check_first_name_only]
   end
 
   def dqt_record(trn, nino)
-    if FeatureFlag.active?(:full_dqt_api)
-      object = full_dqt_client.get_record(trn: trn, birthdate: date_of_birth, nino: nino)
-      FullDqt::OldRecordMapper.translate(object)
```
Comment (kicferk1) [inline app/services/participant_validation_service.rb:56]: Should we cut out `FullDqt::OldRecordMapper` too?

**Review by kicferk1 — COMMENTED** [review]
Is most of the changes in test caused by switching to only using new dqt api?

File: app/services/participant_validation_service.rb (hunk 2 above)
Comment (ethanmills) [reply app/services/participant_validation_service.rb:56]: Yep, good spot. Had intended to

**Comment by ethanmills** [comment]
> Is most of the changes in test caused by switching to only using new dqt api?

Yes mostly. There are a couple of new tests on the induction start date logic, but mostly just changing the stubbing

**Comment by ethanmills** [comment]
We have induction start date in prod now, so this is unblocked

**Review by kicferk1 — APPROVED** [review]

File: app/services/participant_validation_service.rb [hunk 3]
```diff
@@ -19,75 +19,60 @@ def validate
...
-    validation_data
+    validation_data["induction"]["start_date"] < Date.new(Cohort.current.start_year, 9, 1)
   end
 
   def check_first_name_only?
     config[:check_first_name_only]
   end
 
   def dqt_record(trn, nino)
-    if FeatureFlag.active?(:full_dqt_api)
```
Comment (asmega) [inline app/services/participant_validation_service.rb:54]: not important but we should able to remove the flag itself, can go in a later patch. Theres also a boat load of tests that can also be go as I just copied the tests, one with the flag and one without.

Comment (asmega) [reply app/services/participant_validation_service.rb:54]: looks like you already removed the tests 👍

**Review by asmega — APPROVED** [review]

File: app/services/participant_validation_service.rb (hunk 3 above)
Comment (ethanmills) [reply app/services/participant_validation_service.rb:54]: Yep, I'll remove the dependency on the flag, release, deactivate the flag, then remove from the code

**Pull request merged**